sequence_task_1/student/include/impl.h

```diff
@@ -3,39 +3,82 @@
 #include <string>
 #include <stdexcept>
 
+
+
 template<class Container, class Song_t>
 class StaticPlaylist
 {
 public:
     /** @todo Member traits */
+    using value_type = typename Container::value_type;
+    using reference = typename Container::reference;
+    using const_reference = typename Container::const_reference;
+    using iterator = typename Container::iterator;
+    using const_iterator = typename Container::const_iterator;
+    using difference_type = typename Container::difference_type;
+    using size_type = typename Container::size_type;
 
     /** @todo Iterators */
+      const_iterator begin() const{
+        return m_tracklist.cbegin();
+    };
+    const_iterator end() const{
+        return m_tracklist.cend();
+    };
 
     StaticPlaylist() = default;
 
     /** @todo Constructor from any reversible sequence container */
+ 
+    template<class OtherContainer> StaticPlaylist(const OtherContainer & tracklist){
+       m_tracklist.assign(tracklist.rbegin(),tracklist.rend());
```

Comment:
can we use constructor of tracklist here?

Reply:
Yes in this way: 
template<class OtherContainer> StaticPlaylist(const OtherContainer & tracklist):
    m_tracklist(tracklist.rbegin(),tracklist.rend()){}